src/hooks/useArweave.ts

```diff
@@ -0,0 +1,70 @@
+import Arweave from 'arweave';
+import { decrypt } from 'ecies-geth';
+import { utils } from 'ethers';
+import { useNetworkConfig } from 'lib/config';
+import { useCallback } from 'react';
+
+export function useArweave() {
+  const { arweaveConfig } = useNetworkConfig();
+  const arweave = arweaveConfig.host ? Arweave.init(arweaveConfig) : undefined;
```

Comment:
should this maybe be in a `useEffect` with `arweaveConfig` as a dependency? And then set arweave as a local state var?

Reply:
Hmm yes you're right